source/extensions/retry/host/omit_host_metadata/omit_host_metadata.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "envoy/upstream/retry.h"
+#include "envoy/upstream/upstream.h"
+
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Retry {
+namespace Host {
+
+class OmitHostsRetryPredicate : public Upstream::RetryHostPredicate {
+public:
+  explicit OmitHostsRetryPredicate(
+      const envoy::config::core::v3alpha::Metadata metadata_match_criteria)
```

Comment:
you want this to be passed by const ref, not just const. ie `const Metadata&`

Reply:
Done.